composer/loggers/console_logger.py

```diff
@@ -117,10 +127,32 @@ def predict_start(self, state: State, logger: Logger) -> None:
             self._log_hparams_to_console()
 
     def eval_start(self, state: State, logger: Logger) -> None:
+        total_eval_batches = self._get_total_eval_batches(state)
+        deciles = np.linspace(0, 1, NUM_EVAL_LOGGING_EVENTS + 1)
+        batch_idxs = np.arange(1, total_eval_batches + 1)
+        if total_eval_batches < NUM_EVAL_LOGGING_EVENTS:
+            self.eval_batch_idxs_to_log = list(batch_idxs)
+        else:
+            self.eval_batch_idxs_to_log = list(np.quantile(batch_idxs, deciles).round().astype(dtype=int))
```

Comment:
oh just realized one thing. This should be a set I think? Otherwise you duplicate the last print line

Reply:
because of the eval_end event?

Reply:
yeah, eg in your example logs
```
[Eval batch=320/320] Eval on eval data
[Eval batch=320/320] Eval on eval data:
	 Eval LanguageCrossEntropy: 10.9731
	 Eval Perplexity: 58297.7109
	 ```

Reply:
oh right I think maybe just `eval_batch_idxs.remove(num_eval_batches)` perhaos?

Reply:
why not make it a set? It is logically a set, and also the 'in' operation will be (ever so slightly) faster

Reply:
but the issue isn't that `self.eval_batch_idxs_to_log` has duplicate values. It's that `eval_batch_end` runs at the end of the last batch and then `eval_end` runs right after

Reply:
so if we remove the last element of the list then `eval_batch_end` won't run at the end of the last batch

Reply:
ok thanks good catch fixed